skbio/io/format/genbank.py

```diff
@@ -0,0 +1,865 @@
+    seq, md, pmd = record
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            # RNA mol type has T instead of U for genbank from from NCBI
+            constructor = DNA
+        elif unit == 'aa':
+            constructor = Protein
+        else:
+            constructor = Sequence
+    return constructor(
+        seq, metadata=md, positional_metadata=pmd, **kwargs)
+
+
+def _parse_genbanks(fh):
+    data_chunks = []
+    for line in _line_generator(fh, skip_blanks=True, strip=False):
+        if line.startswith('//'):
+            yield _parse_single_genbank(data_chunks)
+            data_chunks = []
+        else:
+            data_chunks.append(line)
+
+
+def _parse_single_genbank(chunks):
+    metadata = {}
+    positional_metadata = None
+    sequence = ''
+    # each section starts with a HEADER without indent.
+    section_splitter = _yield_section(
+        lambda x: not x[0].isspace(), strip=False)
+    for section in section_splitter(chunks):
+        header = section[0].split(None, 1)[0]
+        parser = _PARSER_TABLE.get(
+            header, _parse_section_default)
+
+        if header == 'FEATURES':
+            # This requires 'LOCUS' line parsed before 'FEATURES', which should
+            # be true and is implicitly checked by the sniffer.
+            parser = partial(
+                parser, length=metadata['LOCUS']['size'])
+
+        try:
+            parsed = parser(section)
+        except:
+            raise GenBankFormatError(
+                'Could not parse this section with %s:\n%s' %
+                (parser, ''.join(section)))
+
+        # reference can appear multiple times
+        if header == 'REFERENCE':
+            if header in metadata:
+                metadata[header].append(parsed)
+            else:
+                metadata[header] = [parsed]
+        elif header == 'ORIGIN':
+            sequence = parsed
+        elif header == 'FEATURES':
+            metadata[header] = parsed[0]
+            positional_metadata = pd.concat(parsed[1], axis=1)
+        else:
+            metadata[header] = parsed
+    return sequence, metadata, positional_metadata
+
+
+def _serialize_single_genbank(obj, fh):
+    '''Write a GenBank record.
+
+    Always write it in NCBI canonical way:
+    1. sequence in lowercase
+    2. 'u' as 't' even in RNA molecules.
+    '''
+    md = obj.metadata
+    for header in _HEADERS:
+        if header in md:
+            serializer = _SERIALIZER_TABLE.get(
+                header, _serialize_section_default)
+            out = serializer(header, md[header])
+            # test if 'out' is a iterator.
+            # cf. Effective Python Item 17
+            if iter(out) is iter(out):
+                for s in out:
+                    fh.write(s)
+            else:
+                fh.write(out)
+    # always write RNA seq as DNA
+    if isinstance(obj, RNA):
```

Comment:
We intend to add a `Final` metaclass to `RNA`, `DNA`, and `Protein` in the future as they are not safe to subclass (as any change in their alphabet could alter the functionality of certain methods). So in this case you can just use an `is` check.